Lib/test/support/_interpreters.py

```diff
@@ -0,0 +1,235 @@
+
+    Get all open channels.
+    """
+    return [(RecvChannel(cid), SendChannel(cid))
+            for cid in _interpreters.channel_list_all()]
+
+def wait(timeout):
+    #The implementation for wait
+    # will be non trivial to be useful
+    import time
+    time.sleep(timeout)
+
+class RecvChannel:
+
+    def __init__(self, id):
+        self.id = id
+
+    def recv(self):
+        """ channel_recv() -> obj
+
+        Get the next object from the channel,
+        and wait if none have been sent.
+        Associate the interpreter with the channel.
+        """
+        try:
+            obj = _interpreters.channel_recv(self.id)
+            if obj == None:
+                wait(2)
+                obj = _interpreters.channel_recv(self.id)
+        except _interpreters.ChannelEmptyError:
+            raise ChannelEmptyError
+        except _interpreters.ChannelNotFoundError:
+            raise ChannelNotFoundError
+        except _interpreters.ChannelClosedError:
+            raise ChannelClosedError
+        except _interpreters.RunFailedError:
+            raise RunFailedError
+        return obj
+
+    def recv_nowait(self, default=None):
+        """recv_nowait(default=None) -> object
+
+        Like recv(), but return the default
+        instead of waiting.
+        """
+        try:
+            obj = _interpreters.channel_recv(self.id)
+        except _interpreters.ChannelEmptyError:
+            raise ChannelEmptyError
+        except _interpreters.ChannelNotFoundError:
+            raise ChannelNotFoundError
+        except _interpreters.ChannelClosedError:
+            raise ChannelClosedError
+        except _interpreters.RunFailedError:
+            raise RunFailedError
+        return obj
+
+
+class SendChannel:
+
+    def __init__(self, id):
+        self.id = id
+
+    def send(self, obj):
+        """ send(obj)
+
+        Send the object (i.e. its data) to the receiving
+        end of the channel and wait. Associate the interpreter
+        with the channel.
+        """
+        try:
+            _interpreters.channel_send(self.id, obj)
+            wait(2)
+        except _interpreters.ChannelNotFoundError:
+            raise ChannelNotFoundError
+        except _interpreters.ChannelClosedError:
+            raise ChannelClosedError
+        except _interpreters.RunFailedError:
+            raise RunFailedError
+
+    def send_nowait(self, obj):
+        """ send_nowait(obj)
+
+        Like send(), but return False if not received.
+        """
+        try:
+            _interpreters.channel_send(self.id, obj)
+        except _interpreters.ChannelNotFoundError:
+            raise ChannelNotFoundError
+        except _interpreters.ChannelClosedError:
+            raise ChannelClosedError
+        except _interpreters.RunFailedError:
+            raise RunFailedError
+
+        recv_obj = _interpreters.channel_recv(self.id)
+        if recv_obj:
+            return obj
+        else:
+            return False
```

Comment:
The recv doesn't really fit.  Until #19829 lands you can't return anything so you might as well always return `False`:

```suggestion
        _interpreters.channel_send(self.id, obj)
        return False
```

Once #19829 lands this code will look like this:

```suggestion
        return _interpreters.channel_send(self.id, obj)
```

or:

```suggestion
        wait = _interpreters.channel_send_wait(self.id, obj)
        return wait(timeout=0)
```